### PR TITLE
TDE on BigAnimal with Customer Managed Keys

### PR DESCRIPTION
**Pull request by gvasquezvargas**
## What Changed?
 
https://enterprisedb.atlassian.net/browse/DOCS-724 
Final UI changes are being made as we speak.

### CODE REVIEW

File: product_docs/docs/biganimal/release/administering_cluster/projects.mdx [hunk 1]
```diff
@@ -88,18 +88,19 @@ To delete a project that you created:
...
-   - Select **Next**
-   - Enter AWS Key Management System ARN
-   - Enter a friendly name for your key
-   - Select **Finish**
+1. On the **Add a key** page, select the **Cloud Service Provider**. 
+1. Select the **Region** for the key. The interface only displays the regions available in the cloud account you configured.
+   !!! Note 
+   GCP offers the option to configure global keys. If you require a global key, do not select any region.
+   !!! 
+1. Complete the remaining fields according to your cloud provider.
```
Comment (gvasquezvargas) [inline product_docs/docs/biganimal/release/administering_cluster/projects.mdx:96]: I don't think we need provider-specific instructions here, as the UI is very clear. 

The only "difficult" thing is obtaining the key version for the Azure config. We can consider adding a hint/note.

File: product_docs/docs/biganimal/release/getting_started/creating_a_cluster/index.mdx [hunk 2]
```diff
@@ -213,16 +214,82 @@ Enable **Superuser Access** to grant superuser privileges to the edb_admin role.
...
-- If the TDE enabled cluster creation is in progress and the process is waiting providing **Encryption Key Error** next to the cluster name on the clusters page:
-  - Select the cluster name and go to the cluster's home page.
-  - See the **Action required: grant key permissions to activate the cluster** on the cluster's overview tab.
-  - Copy the **Principal** and add it to your key policy to provide **kms:Encrypt** and **kms:Decrypt** permission. 
+- To enable and use TDE for a cluster, the encryption key must be enabled and added at the project level before creating a cluster. 
+  To add a key, see [Adding a TDE key at project level](../../administering_cluster/projects.mdx/#adding-a-tde-key).
+- To enable and use TDE for a cluster, you must complete the configuration on the platform of your key management provider after creating a cluster. See [Completing the TDE configuration](#completing-the-TDE-configuration) for more information.
 !!!
 
+#### Completing the TDE configuration
```
Comment (gvasquezvargas) [inline product_docs/docs/biganimal/release/getting_started/creating_a_cluster/index.mdx:225]: This is option 1, with the full instructions per cloud provider on how to finalize the configuration.

File: product_docs/docs/biganimal/release/getting_started/creating_a_cluster/index.mdx [hunk 3]
```diff
@@ -213,16 +214,82 @@ Enable **Superuser Access** to grant superuser privileges to the edb_admin role.
...
+    1. Got to the Microsoft Azure console, and navigate to **Key vaults**. 
+    1. Select the key, **Access policies**, and **Create**.
+    1. In **Permissions**, select **Encrypt** and **Decrypt**.
+    1. In **Principal**, paste the MSI Workload Identity you copied to your clipboard and finish creating the policy.
+
+    </details>
+    
+
+
+#### Completing the TDE configuration 2
```
Comment (gvasquezvargas) [inline product_docs/docs/biganimal/release/getting_started/creating_a_cluster/index.mdx:264]: This is option 2, with simplified instructions per cloud provider on how to finalize the configuration. 
It is less accurate, but we would not have to worry about the instructions becoming outdated without us noticing.

File: product_docs/docs/biganimal/release/overview/03_security/index.mdx [hunk 4]
```diff
@@ -1,28 +1,65 @@
...
+
+All data in BigAnimal is encrypted in motion and at rest. Network traffic is encrypted using Transport Layer Security (TLS) v1.2 or greater. Data at rest is encrypted using AES with 256-bit keys. Data encryption keys are envelope encrypted, and the wrapped data encryption keys are securely stored in a key management system. When you use your own cloud account, encryption keys never leave your cloud environment.
+
+### Your own encryption key - Transparent Data Encryption (TDE)
+
+Optionally enable Transparent Data Encryption (TDE) at the database level on BigAnimal's cloud account, on AWS, GCP or Azure. TDE encrypts all data files, the write-ahead log (WAL) and temporary files used during query processing and database system operations. 
+
+You can't enable nor disable TDE on existing clusters. To enable TDE, first connect the encryption keys to BigAnimal at the project level, and then select those keys while creating a cluster. 
+
+Enabling TDE using your own encryption key is supported on EDB Postgres Advanced Server and EDB Postgres Extended Server versions 15 and later. Both the key and the cluster must be within the same region and the same cloud provider:
```
Comment (gvasquezvargas) [inline product_docs/docs/biganimal/release/overview/03_security/index.mdx:32]: Not sure if they have to be on the same cloud provider... To be clarified with dev.

File: product_docs/docs/biganimal/release/overview/03_security/index.mdx [hunk 5]
```diff
@@ -1,28 +1,65 @@
...
+
+### Your own encryption key - Transparent Data Encryption (TDE)
+
+Optionally enable Transparent Data Encryption (TDE) at the database level on BigAnimal's cloud account, on AWS, GCP or Azure. TDE encrypts all data files, the write-ahead log (WAL) and temporary files used during query processing and database system operations. 
+
+You can't enable nor disable TDE on existing clusters. To enable TDE, first connect the encryption keys to BigAnimal at the project level, and then select those keys while creating a cluster. 
+
+Enabling TDE using your own encryption key is supported on EDB Postgres Advanced Server and EDB Postgres Extended Server versions 15 and later. Both the key and the cluster must be within the same region and the same cloud provider: 
+
+|                             | AWS-hosted cluster (BYOA) | GCP-hosted cluster (BYOA) | Azure-hosted cluster (BYOA) | BigAnimal-hosted cluster (BAH) |
```
Comment (gvasquezvargas) [inline product_docs/docs/biganimal/release/overview/03_security/index.mdx:34]: Option 1 if cross-provider support is given.

File: product_docs/docs/biganimal/release/overview/03_security/index.mdx [hunk 6]
```diff
@@ -1,28 +1,65 @@
...
+|-----------------------------|---------------------------|---------------------------|-----------------------------|--------------------------------|
+| AWS Key Management Service  | yes                       | yes                       | yes                         | yes                            |
+| Google Cloud Key Management | yes                       | yes                       | yes                         | yes                            |
+| Azure Key Vault             | yes                       | yes                       | yes                         | no                             |
+
+|                             | AWS-hosted cluster (BYOA) | GCP-hosted cluster (BYOA) | Azure-hosted cluster (BYOA) | BigAnimal-hosted cluster (BAH) |
```
Comment (gvasquezvargas) [inline product_docs/docs/biganimal/release/overview/03_security/index.mdx:40]: Option 2 if cluster and key must be on the same provider.

File: product_docs/docs/biganimal/release/administering_cluster/projects.mdx [hunk 7]
```diff
@@ -88,18 +88,19 @@ To delete a project that you created:
...
-1. On the **Add a key** page, select:
-   - Select the **Provider & Region**
-   - Select **Next**
-   - Enter AWS Key Management System ARN
-   - Enter a friendly name for your key
-   - Select **Finish**
+1. On the **Add a key** page, select the **Cloud Service Provider**. 
+1. Select the **Region** for the key. The interface only displays the regions available in the cloud account you configured.
+   !!! Note 
+   GCP offers the option to configure global keys. If you require a global key, do not select any region.
```
Comment (gvasquezvargas) [inline product_docs/docs/biganimal/release/administering_cluster/projects.mdx:94]: Waiting on UX improvement. It is possible the UI will contain a "global" selector, which will make this note obsolete.

File: product_docs/docs/biganimal/release/getting_started/creating_a_cluster/index.mdx [hunk 8]
```diff
@@ -213,16 +214,54 @@ Enable **Superuser Access** to grant superuser privileges to the edb_admin role.
...
+1.  Copy the **Principal** identifier (AWS), **service account** (GCP) or **MSI Workload Identity** (Azure) to your clipboard. 
+
+1.  Follow the on-screen guide to grant encrypt and decrypt permissions to your key. Here is additional information in case you require further guidance:
+
+    <details><summary>AWS</summary>
+
+    1. Copy the **Principal** identifier to your clipboard. 
+    1. Go to the AWS console, and navigate to the **Key Management Service**
+    1. Select **Customer-managed keys**, and **Edit policy** for your key.
+    1. Paste the **Principal** identifier you copied to your clipboard into the `Principal.AWS` field.
```
Comment (GabriFedi97) [inline product_docs/docs/biganimal/release/getting_started/creating_a_cluster/index.mdx:240]: We should mention that a new statement needs to be added. The Policy should change from something like this:
```
{
    "Version": "2012-10-17",
    "Id": "key-consolepolicy-3",
    "Statement": [
        {
            "Sid": "Enable IAM User Permissions",
            "Effect": "Allow",
            "Principal": {
                "AWS": "arn:aws:iam::142348057180:root"
            },
            "Action": "kms:*",
            "Resource": "*"
        }
    ]
}
```

to something like this:
```
{
    "Version": "2012-10-17",
    "Id": "key-consolepolicy-3",
    "Statement": [
        {
            "Sid": "Enable IAM User Permissions",
            "Effect": "Allow",
            "Principal": {
                "AWS": "arn:aws:iam::142348057180:root"
            },
            "Action": "kms:*",
            "Resource": "*"
        },
        {
            "Sid": "Enable TDE on cluster p-k2nxxtndkg",
            "Effect": "Allow",
            "Principal": {
                "AWS": "arn:aws:iam::142348057180:role/ba-vcvolsnap0ZUPsgZ-us-east-1-p-k2nxxtndkg"
            },
            "Action": [
                "kms:Encrypt",
                "kms:Decrypt"
            ],
            "Resource": "*"
        }
    ]
}
```

File: product_docs/docs/biganimal/release/getting_started/creating_a_cluster/index.mdx [hunk 9]
```diff
@@ -213,16 +214,54 @@ Enable **Superuser Access** to grant superuser privileges to the edb_admin role.
...
+    1. Paste the service account into the **New principals** field.
+    1. Assign the `Cloud KMS CryptoKey Decrypter` and `Cloud KMS CryptoKey Encrypter` roles and save. 
+
+    </details>
+
+    <details><summary>Azure</summary>
+
+    1. Copy the **MSI Workload Identity** to your clipboard. 
+    1. Got to the Microsoft Azure console, and navigate to **Key vaults**. 
+    1. Select the key, **Access policies**, and **Create**.
```
Comment (GabriFedi97) [inline product_docs/docs/biganimal/release/getting_started/creating_a_cluster/index.mdx:258]: We should mention to set the `Permission model` to `Vault access policy` through the `Access Configuration` section

**Review by GabriFedi97 — CHANGES_REQUESTED** [review]

File: product_docs/docs/biganimal/release/using_cluster/managing_replicas.mdx [hunk 10]
```diff
@@ -53,9 +52,9 @@ You can create faraway replicas in any active regions in your cloud. There's no
    Under the **Security** section, the **Transparent Data Encryption (TDE)** option is enabled by default only when your primary cluster is TDE-enabled. It automatically enables TDE and allows you to select the encryption key from the available List.
 
    !!!Note
-   TDE key material for faraway replicas must be same as the primary cluster encryption key. In case you use different key material, the cluster provisioning fails.
-
-   We recommend, not to enable TDE for faraway replica cluster creation, if the source cluster is a non-TDE cluster.
+   The TDE key material for faraway replicas must be the same as the primary cluster encryption key. If you use different key material, the cluster provisioning will fail.
+ 
+   Do not to enable TDE when restoring a non-TDE faraway replica cluster.
```
[the remark this answers is not in the document]
Comment (gvasquezvargas) [reply product_docs/docs/biganimal/release/using_cluster/managing_replicas.mdx:57]: Same here, I was making a stylistic edit.

Do you think we should be more explicit and tell customers this is not supported? I am not sure if the CLI would do any pre-flight checks and prevent customers from trying to restore a non-TDE faraway replica cluster with TDE enabled. So I think _technically_ they could?

File: product_docs/docs/biganimal/release/using_cluster/05a_deleting_your_cluster.mdx [hunk 11]
```diff
@@ -30,7 +30,7 @@ You can restore your deleted cluster for as long as the backup is available.
     When the process completes, the restored cluster is available on the [Clusters](https://portal.biganimal.com/clusters) page.
 
 !!! note
-To restore a TDE enabled cluster, the TDE key material must match with source cluster encryption key material. In case a different key material is used the restore operation fails.
+To restore a TDE-enabled cluster, the TDE key material must match the source cluster encryption key material. If a different key material is used, the restore operation will fail.
 
-We recommend, not to enable TDE while restoring a cluster, if the source cluster is a non-TDE cluster.
+Do not to enable TDE when restoring a non-TDE cluster.
```
[the remark this answers is not in the document]
Comment (GabriFedi97) [reply product_docs/docs/biganimal/release/using_cluster/05a_deleting_your_cluster.mdx:35]: I don't see any value in letting something happen even if we know that will fail for sure, so I think we should prevent that. The UI seems to not be doing that, I'll raise a ticket for the same.

File: product_docs/docs/biganimal/release/using_cluster/managing_replicas.mdx (hunk 10 above)
[the remark this answers is not in the document]
Comment (GabriFedi97) [reply product_docs/docs/biganimal/release/using_cluster/managing_replicas.mdx:57]: Same as above

File: product_docs/docs/biganimal/release/using_cluster/05a_deleting_your_cluster.mdx (hunk 11 above)
[the remark this answers is not in the document]
Comment (gvasquezvargas) [reply product_docs/docs/biganimal/release/using_cluster/05a_deleting_your_cluster.mdx:35]: @GabriFedi97 gotcha! What do you think we should do here in the docs? Maybe something along the lines of: 
```suggestion
EDB does not support enabling TDE when restoring a non-TDE cluster.
```

File: product_docs/docs/biganimal/release/using_cluster/managing_replicas.mdx (hunk 10 above)
[the remark this answers is not in the document]
Comment (gvasquezvargas) [reply product_docs/docs/biganimal/release/using_cluster/managing_replicas.mdx:57]: And here we could say: 
```suggestion
   EDB does not support enabling TDE when restoring non-TDE faraway replica clusters.
```

File: product_docs/docs/biganimal/release/getting_started/creating_a_cluster/index.mdx [hunk 12]
```diff
@@ -213,16 +214,86 @@ Enable **Superuser Access** to grant superuser privileges to the edb_admin role.
...
+                         "AWS": "arn:aws:iam::142348057180:root"
+                     },
+                     "Action": "kms:*",
+                     "Resource": "*"
+                 },
+                 {
+                     "Sid": "Enable TDE on cluster ExampleCluster",
+                     "Effect": "Allow",
+                     "Principal": {
+                         "AWS": "arn:aws:iam::142348057180:role/ba-vcvolsnap0ZUPsgZ-us-east-1-p-k2nxxtndkg"
```
Comment (GabriFedi97) [inline product_docs/docs/biganimal/release/getting_started/creating_a_cluster/index.mdx:262]:
```suggestion
                         "AWS": "arn:aws:iam::<aws_project_id>:role/<pg_cluster_role>"
```

File: product_docs/docs/biganimal/release/getting_started/creating_a_cluster/index.mdx [hunk 13]
```diff
@@ -213,16 +214,86 @@ Enable **Superuser Access** to grant superuser privileges to the edb_admin role.
...
+        ```
+        {
+             "Version": "2012-10-17",
+             "Id": "key-consolepolicy-3",
+             "Statement": [
+                 {
+                     "Sid": "Enable IAM User Permissions",
+                     "Effect": "Allow",
+                     "Principal": {
+                         "AWS": "arn:aws:iam::142348057180:root"
```
Comment (GabriFedi97) [inline product_docs/docs/biganimal/release/getting_started/creating_a_cluster/index.mdx:253]:
```suggestion
                         "AWS": "arn:aws:iam::<aws_project_id>:root"
```

**Review by GabriFedi97 — CHANGES_REQUESTED** [review]

**Review by nidhibhammar — APPROVED** [review]
LGTM

**Review by GabriFedi97 — APPROVED** [review]

File: product_docs/docs/biganimal/release/administering_cluster/projects.mdx [hunk 14]
```diff
@@ -88,18 +88,21 @@ To delete a project that you created:
...
+   !!! 
+1. Complete the remaining fields according to your cloud provider. 
+1. Select **Add Key** to finalize the configuration.
 
-Now, use this TDE key to create a cluster. For more information, see [Creating a cluster](/biganimal/release/getting_started/creating_a_cluster/#security).
+Now, use this TDE key to [create a cluster](/biganimal/release/getting_started/creating_a_cluster/#security).
```
Comment (gvasquezvargas) [inline product_docs/docs/biganimal/release/administering_cluster/projects.mdx:99]: Add a note above this: 

Note for GCP keys:
If the key you added was created in a different Google Cloud Platform account than the cluster, ensure you enable the [Cloud KMS API](https://cloud.google.com/kms/docs/create-encryption-keys#before-you-begin) in the Google console before assigning it to your cluster in BigAnimal.

**Comment by dbwagoner** [comment]
I read the new TDE documentation and it looks good to me! Thank you!

**Pull request merged**